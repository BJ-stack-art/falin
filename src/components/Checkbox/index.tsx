import React, { useId, useState } from "react";
import CheckedIcon from "@/assets/forms/checkbox-checked.svg";
import HoverIcon from "@/assets/forms/checkbox-hover.svg";
import UncheckIcon from "@/assets/forms/checkbox-uncheck.svg";
import {
  Box,
  FormControlLabel,
  Checkbox as MuiCheckbox,
  SxProps,
  FormControlLabelProps,
} from "@mui/material";

const styles: { box: SxProps; label: SxProps; checkbox: SxProps } = {
  box: { marginLeft: "7px" },
  label: {
    "& span.MuiFormControlLabel-label": {
      marginLeft: "10px",
      fontWeight: 500,
      fontSize: { xs: "14px", md: "16px" },
    },
  },
  checkbox: {
    position: "relative",
    marginLeft: "5px",
    marginTop: "0px",
    "&:disabled": {
      pointerEvents: "none",
    },
  },
};

export interface ICheckboxProps
  extends Omit<FormControlLabelProps, keyof { control: any }> {
  align?: "center" | "top";
}
const Checkbox: React.FC<ICheckboxProps> = ({ align, ...props }) => {
  const id = useId();
  const [icon, setIcon] = useState(UncheckIcon);

  const controlLabelSx = {
    alignItems: align === "center" ? "center" : "flex-start",
    "& span.MuiCheckbox-root": {
      marginTop: align === "center" ? "0px" : "4px",
    },
    ...styles.label,
  };

  return (
    <Box data-testid="checkbox-label" id={id} sx={styles.box}>
      <FormControlLabel
        {...props}
        sx={controlLabelSx}
        onMouseOver={() => !props.disabled && setIcon(HoverIcon)}
        onMouseLeave={() => setIcon(UncheckIcon)}
        control={
          <MuiCheckbox
            sx={styles.checkbox}
            icon={<img style={{ position: "absolute" }} src={icon} />}
            checkedIcon={
              <img style={{ position: "absolute" }} src={CheckedIcon} />
            }
          />
        }
      />
    </Box>
  );
};

export default Checkbox;
