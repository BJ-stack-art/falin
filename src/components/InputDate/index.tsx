import { FormControl, FormHelperText, InputLabel } from "@mui/material"
import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs"
import { DatePicker, DatePickerProps } from "@mui/x-date-pickers/DatePicker"
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider"
import { Dayjs } from "dayjs"
import { CalendarBlank } from "phosphor-react"
import React, { useId } from "react"
import "./_.scss"

export interface IInputDateProps extends DatePickerProps<Dayjs> {
  label: string
  helperText?: string
  required?: boolean
  error?: boolean
}
const InputDate: React.FC<IInputDateProps> = (props) => {
  const id = useId()
  const { label, required, error, helperText, ...datePickerProps } = props

  return (
    <FormControl id={id} fullWidth required={required} error={error} disabled={props.disabled}>
      <InputLabel>{label}</InputLabel>
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <DatePicker
          format="DD/MM/YYYY"
          className={`date-picker ${error && "Mui-error"}`}
          views={["year", "month", "day"]}
          slots={{
            openPickerIcon: CalendarBlank,
          }}
          slotProps={{
            popper: {
              className: "date-picker__popper",
            },
          }}
          {...datePickerProps}
        />
      </LocalizationProvider>
      <FormHelperText>{helperText}</FormHelperText>
    </FormControl>
  )
}

export default InputDate
