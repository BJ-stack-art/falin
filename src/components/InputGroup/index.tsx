import { FormControl, FormHelperText, Input, InputLabel, InputProps } from "@mui/material"
import React, { useId } from "react"

export interface IInputGroup extends InputProps {
  label: string
  helperText?: string
}
const InputGroup: React.FC<IInputGroup> = (props) => {
  const id = useId()
  const { label, helperText, ...inputProps } = props
  return (
    <FormControl fullWidth required={props.required} error={props.error} disabled={props.disabled}>
      <InputLabel htmlFor={id}>{label}</InputLabel>
      <Input id={id} {...inputProps} />
      <FormHelperText>{helperText}</FormHelperText>
    </FormControl>
  )
}

export default InputGroup
