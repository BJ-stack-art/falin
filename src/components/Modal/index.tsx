export type { IModalHeaderProps } from "./resources/Header"
export type { IModalBodyProps } from "./resources/Body"
export type { IModalProps } from "./resources/Modal"
export type { IModalFooterProps } from "./resources/Footer"
import Modal from "./resources/Modal"
export { Modal }

export type { IModalErrorProps } from "./resources/ModalError"
import ModalError from "./resources/ModalError"
export { ModalError }
