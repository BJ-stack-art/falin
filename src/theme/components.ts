import { Components } from "@mui/material";
import { danger, info, neutral, primary, success, warning } from "./colors";
import { shadows } from "./shadows";

declare module "@mui/material/Button" {
  interface ButtonPropsSizeOverrides {
    small: false;
    medium: false;
    large: false;
    sm: true;
    md: true;
    lg: true;
    xl: true;
    "2xl": true;
  }
}
declare module "@mui/material/IconButton" {
  interface IconButtonPropsSizeOverrides {
    small: false;
    medium: false;
    large: false;
    sm: true;
    md: true;
    lg: true;
    xl: true;
    "2xl": true;
  }
}

const objColor = {
  primary: primary,
  neutral: neutral,
  success: success,
  info: info,
  warning: warning,
  danger: danger,
};

const fnBtnColorStyle = (
  color: "primary" | "neutral" | "success" | "info" | "warning" | "danger",
  variant: "contained" | "text"
) => {
  if (variant === "contained")
    return {
      color: "#fff",
      background: objColor[color][600],
      ":hover": {
        background: objColor[color][700],
      },
      ":focus": {
        boxShadow: `0px 0px 0px 4px ${objColor[color][100]}`,
      },
      ":disabled": {
        background: objColor[color][200],
        color: "#fff",
      },
    };
  if (variant === "text")
    return {
      background: objColor[color][50],
      ":hover": {
        background: objColor[color][100],
      },
      ":focus": {
        background: objColor[color][50],
        boxShadow: `0px 0px 0px 4px ${objColor[color][100]}`,
      },
      ":disabled": {
        background: "none",
        color: objColor[color][300],
      },
    };
};

export const components: Components = {
  MuiButton: {
    defaultProps: {
      variant: "contained",
      size: "md",
      disableRipple: true,
    },
    styleOverrides: {
      root: {
        borderRadius: "8px",
        textTransform: "capitalize",
        fontWeight: 600,

        "&[data-shape='icon']": {
          minWidth: "0px",
          padding: "0px !important",
          "& span.MuiCircularProgress-root": {
            height: "18px !important",
            width: "18px !important",
          },
        },

        "& span.MuiCircularProgress-root": {
          height: "24px !important",
          width: "24px !important",
        },

        // --- size sm ---
        "&.MuiButton-sizeSm": {
          height: "36px",
          fontSize: "14px",
          padding: "8px 14px",
          "&[data-shape='icon']": {
            height: "36px",
            width: "36px",
            "& svg": {
              width: "17px",
            },
          },
        },
        // --- size md ---
        "&.MuiButton-sizeMd": {
          height: "40px",
          fontSize: "14px",
          padding: "10px 16px",
          "&[data-shape='icon']": {
            height: "40px",
            width: "40px",
            "& svg": {
              width: "17px",
            },
          },
        },
        // --- size lg ---
        "&.MuiButton-sizeLg": {
          height: "44px",
          fontSize: "16px",
          padding: "10px 18px",
          "&[data-shape='icon']": {
            height: "44px",
            width: "44px",
            "& svg": {
              width: "17px",
            },
          },
        },
        // --- size xl ---
        "&.MuiButton-sizeXl": {
          height: "48px",
          fontSize: "16px",
          padding: "12px 20px",
          "&[data-shape='icon']": {
            height: "48px",
            width: "48px",
            "& svg": {
              width: "17px",
            },
          },
        },
        // --- size 2xl ---
        "&.MuiButton-size2xl": {
          height: "60px",
          fontSize: "18px",
          padding: "16px 28px",
          "&[data-shape='icon']": {
            height: "56px",
            width: "56px",
            "& svg": {
              width: "20px",
            },
          },
        },
      },

      containedPrimary: fnBtnColorStyle("primary", "contained"),
      textPrimary: fnBtnColorStyle("primary", "text"),
      containedInfo: fnBtnColorStyle("info", "contained"),
      textInfo: fnBtnColorStyle("info", "text"),
      containedSuccess: fnBtnColorStyle("success", "contained"),
      textSuccess: fnBtnColorStyle("success", "text"),
      containedWarning: fnBtnColorStyle("warning", "contained"),
      textWarning: fnBtnColorStyle("warning", "text"),
      containedError: fnBtnColorStyle("danger", "contained"),
      textError: fnBtnColorStyle("danger", "text"),
      containedInherit: fnBtnColorStyle("neutral", "contained"),
      textInherit: fnBtnColorStyle("neutral", "text"),
    },
  },
  MuiCircularProgress: {
    defaultProps: {
      disableShrink: true,
      color: "inherit",
    },
  },
  MuiCard: {
    styleOverrides: {
      root: {
        padding: "20px 16px",
        boxShadow: shadows.lg,
      },
    },
  },
  MuiFormControl: {
    styleOverrides: {
      root: {
        marginTop: "16px",
        "& .MuiFormHelperText-root": {
          transform: "translate(-14px)",
          fontSize: "14px",
          marginTop: "6px",
        },
      },
    },
  },
  MuiInputLabel: {
    defaultProps: {
      shrink: true,
    },
    styleOverrides: {
      root: {
        marginBottom: "4px",
        color: "inherit",
        "& .MuiFormLabel-asterisk": {
          color: danger[400],
        },
      },
      shrink: {
        fontSize: "14px",
        transform: "scale(1) translate(0px, -8px)",
      },
    },
  },
  MuiInput: {
    defaultProps: {
      disableUnderline: true,
    },
    styleOverrides: {
      root: {
        "&:not([datatype='select-input'])": {
          backgroundColor: "#fff",
          border: `1px solid ${neutral[300]}`,
          borderRadius: "8px",
          height: "40px",
          padding: "8px 12px",
          width: "100%",
          transition: ".25s",
          "& ::placeholder": {
            color: neutral[500],
          },
          "&.Mui-focused": {
            border: `1px solid ${primary[300]}`,
            boxShadow: `0px 0px 0px 3px ${primary[100]}`,
          },
        },
        "&.Mui-error": {
          border: `1px solid ${danger[300]}`,
          "& svg": {
            color: danger[500],
          },
          "&.Mui-focused": {
            border: `1px solid ${danger[300]}`,
            boxShadow: `0px 0px 0px 3px ${danger[100]}`,
          },
        },
        "&.MuiInputBase-multiline": {
          height: "auto",
        },
      },
    },
  },
  MuiFormControlLabel: {
    styleOverrides: {
      root: {
        "&.MuiFormControlLabel-label": {
          fontWeight: 400,
        },
      },
    },
  },
};
