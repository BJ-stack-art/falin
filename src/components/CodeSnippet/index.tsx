import useShowCode from "@/hooks/useShowCode";
import { neutral } from "@/theme/colors";
import { Collapse } from "@mui/material";
import React from "react";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { materialLight } from "react-syntax-highlighter/dist/esm/styles/prism";
import BtnShowCode from "../BtnShowCode";

const CodeSnippet: React.FC<{ code: string }> = ({ code }) => {
  const inputState = useShowCode();

  return (
    <>
      <BtnShowCode onClick={inputState.toggle} />
      <Collapse in={inputState.show}>
        <SyntaxHighlighter
          customStyle={{
            width: "100%",
            border: `1px solid ${neutral[300]}`,
            borderRadius: "8px",
            maxHeight: "250px",
            fontSize: "12px",
          }}
          showLineNumbers
          language="tsx"
          wrapLines
          style={materialLight}
        >
          {code}
        </SyntaxHighlighter>
      </Collapse>
    </>
  );
};

export default CodeSnippet;
