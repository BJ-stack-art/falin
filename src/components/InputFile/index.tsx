import { appendStyle } from "@/utils/styles"
import { Box, Button, Chip, FormControl, FormHelperText, FormLabel, InputLabel, Stack, SxProps } from "@mui/material"
import React, { useEffect, useId, useRef, useState } from "react"
import { danger, neutral } from "@/theme/colors"

interface IStyle {
  root: SxProps
  error: SxProps
  disabled: SxProps
  label: SxProps
  btn: SxProps
}

const styles: IStyle = {
  root: {
    backgroundColor: "#fff",
    borderRadius: "8px",
    border: `1px solid ${neutral[300]}`,
    boxShadow: "0px 2px 2px rgba(218, 227, 225, 0.1), inset 0px -1px 1px rgba(218, 227, 225, 0.15)",
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
    padding: "0px",
    paddingLeft: "12px",
    cursor: "pointer",
    overflow: "hidden",
    margin: "6px auto",
    marginTop: "15px",
    marginBottom: "0px",
    height: "40px",
  },
  error: {
    border: `1px solid ${danger[300]}`,
  },
  disabled: {
    pointerEvents: "none",
  },
  label: {
    pointerEvents: "none",
    fontSize: "16px",
    marginTop: "4px",
  },
  btn: {
    boxShadow: "none",
    color: "#fff",
    borderRadius: "0px",
  },
}

export interface IInputFile extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "onChange"> {
  onChange?: (value: File | FileList | File[]) => void
  label: string
  helperText?: string
  required?: boolean
  error?: boolean
  disabled?: boolean
  placeholder?: string
  color?: "inherit" | "primary" | "error" | "secondary" | "info" | "success" | "warning"
}
const InputFile: React.FC<IInputFile> = React.forwardRef(
  (
    {
      label,
      helperText,
      required,
      error,
      disabled,
      placeholder = "Select a file...",
      color = "primary",
      onChange,
      ...props
    },
    ref
  ) => {
    const id = useId()
    const inputRef = useRef<HTMLInputElement | null>(null)
    const [fileName, setFileName] = useState<string | null>(null)
    const [file, setFile] = useState<File | null>(null)
    const [files, setFiles] = useState<FileList | File[]>([])

    useEffect(() => {
      if (file && onChange && !props.multiple) {
        onChange(file)
      }
    }, [file])

    useEffect(() => {
      if (files && onChange && props.multiple) {
        onChange(files)
      }
    }, [files])

    const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
      if (e.target.files) {
        if (!props.multiple) {
          setFileName(e.target.files[0].name)
          setFile(e.target.files[0])
        } else {
          setFiles(e.target.files)
        }
      }
    }

    const Placeholder = () => (
      <FormLabel
        sx={[...appendStyle(styles.label), { color: !fileName ? neutral[400] : "inherit" }]}
        required={false}
        error={false}
      >
        {props.multiple ? placeholder : fileName || placeholder}
      </FormLabel>
    )

    const InputBox = () => (
      <>
        <input id={id} {...props} disabled={disabled} ref={inputRef} type="file" hidden onChange={handleChangeInput} />
        <label htmlFor={id}>
          <Box sx={[...appendStyle(styles.root), error && styles.error, disabled && styles.disabled]}>
            <Placeholder />
            <Button color={color} sx={styles.btn} disabled={disabled} onClick={() => inputRef?.current?.click()}>
              Browse file
            </Button>
          </Box>
        </label>
      </>
    )

    const ListFiles = () => (
      <Stack direction="row" flexWrap={"wrap"} spacing={1} rowGap={1} sx={{ marginTop: "4px" }}>
        {Array.from(files).map((item) => (
          <Chip
            key={item.name}
            size="small"
            label={item.name}
            onDelete={() => setFiles(Array.from(files).filter((file) => file.name !== item.name))}
          />
        ))}
      </Stack>
    )

    return (
      <Box ref={ref}>
        <FormControl required={required} disabled={disabled} error={error}>
          <InputLabel htmlFor={id}>{label}</InputLabel>
          <InputBox />
          <FormHelperText>{helperText}</FormHelperText>
          {props.multiple && <ListFiles />}
        </FormControl>
      </Box>
    )
  }
)

InputFile.displayName = "InputFile"
export default InputFile
