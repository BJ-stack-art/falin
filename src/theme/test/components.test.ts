import { describe, expect, test } from "vitest"
import { components } from "../components"

describe("Theme Component", () => {
  test("should match snapshot", () => {
    expect(components).toMatchSnapshot()
  })
})
