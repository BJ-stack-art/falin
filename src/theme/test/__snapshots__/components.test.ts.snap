// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Theme Component > should match snapshot 1`] = `
{
  "MuiAlert": {
    "styleOverrides": {
      "filled": {
        "color": "#fff",
      },
      "filledError": {
        "background": "#ef4444",
      },
      "filledInfo": {
        "background": "#0ea5e9",
      },
      "filledSuccess": {
        "background": "#12b76a",
      },
      "filledWarning": {
        "background": "#f97316",
      },
      "outlinedError": {
        "border": "1px solid #fca5a5",
        "color": "#b91c1c",
      },
      "outlinedInfo": {
        "border": "1px solid #7dd3fc",
        "color": "#0369a1",
      },
      "outlinedSuccess": {
        "border": "1px solid #6ce9a6",
        "color": "#027a48",
      },
      "outlinedWarning": {
        "border": "1px solid #fdba74",
        "color": "#c2410c",
      },
      "root": {
        "borderRadius": "12px",
      },
    },
  },
  "MuiAlertTitle": {
    "styleOverrides": {
      "root": {
        "fontWeight": 600,
      },
    },
  },
  "MuiButton": {
    "defaultProps": {
      "disableRipple": true,
      "size": "md",
      "variant": "contained",
    },
    "styleOverrides": {
      "containedError": {
        ":disabled": {
          "background": "#fecaca",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #fee2e2",
        },
        ":hover": {
          "background": "#b91c1c",
        },
        "background": "#dc2626",
        "color": "#fff",
      },
      "containedInfo": {
        ":disabled": {
          "background": "#bae6fd",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #e0f2fe",
        },
        ":hover": {
          "background": "#0369a1",
        },
        "background": "#0284c7",
        "color": "#fff",
      },
      "containedInherit": {
        ":disabled": {
          "background": "#e2e8f0",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #f1f5f9",
        },
        ":hover": {
          "background": "#334155",
        },
        "background": "#475569",
        "color": "#fff",
      },
      "containedPrimary": {
        ":disabled": {
          "background": "#bfdbfe",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #dbeafe",
        },
        ":hover": {
          "background": "#1d4ed8",
        },
        "background": "#2563eb",
        "color": "#fff",
      },
      "containedSuccess": {
        ":disabled": {
          "background": "#a6f4c5",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #d1fadf",
        },
        ":hover": {
          "background": "#027a48",
        },
        "background": "#039855",
        "color": "#fff",
      },
      "containedWarning": {
        ":disabled": {
          "background": "#fed7aa",
          "color": "#fff",
        },
        ":focus": {
          "boxShadow": "0px 0px 0px 4px #ffedd5",
        },
        ":hover": {
          "background": "#c2410c",
        },
        "background": "#ea580c",
        "color": "#fff",
      },
      "root": {
        "& span.MuiCircularProgress-root": {
          "height": "24px !important",
          "width": "24px !important",
        },
        "&.MuiButton-size2xl": {
          "&[data-shape='icon']": {
            "& svg": {
              "width": "20px",
            },
            "height": "56px",
            "width": "56px",
          },
          "fontSize": "18px",
          "height": "60px",
          "padding": "16px 28px",
        },
        "&.MuiButton-sizeLg": {
          "&[data-shape='icon']": {
            "& svg": {
              "width": "17px",
            },
            "height": "44px",
            "width": "44px",
          },
          "fontSize": "16px",
          "height": "44px",
          "padding": "10px 18px",
        },
        "&.MuiButton-sizeMd": {
          "&[data-shape='icon']": {
            "& svg": {
              "width": "17px",
            },
            "height": "40px",
            "width": "40px",
          },
          "fontSize": "14px",
          "height": "40px",
          "padding": "10px 16px",
        },
        "&.MuiButton-sizeSm": {
          "&[data-shape='icon']": {
            "& svg": {
              "width": "17px",
            },
            "height": "36px",
            "width": "36px",
          },
          "fontSize": "14px",
          "height": "36px",
          "padding": "8px 14px",
        },
        "&.MuiButton-sizeXl": {
          "&[data-shape='icon']": {
            "& svg": {
              "width": "17px",
            },
            "height": "48px",
            "width": "48px",
          },
          "fontSize": "16px",
          "height": "48px",
          "padding": "12px 20px",
        },
        "&[data-shape='icon']": {
          "& span.MuiCircularProgress-root": {
            "height": "18px !important",
            "width": "18px !important",
          },
          "minWidth": "0px",
          "padding": "0px !important",
        },
        "borderRadius": "8px",
        "fontWeight": 600,
        "textTransform": "capitalize",
      },
      "textError": {
        ":disabled": {
          "background": "none",
          "color": "#fca5a5",
        },
        ":focus": {
          "background": "#fef2f2",
          "boxShadow": "0px 0px 0px 4px #fee2e2",
        },
        ":hover": {
          "background": "#fee2e2",
        },
        "background": "#fef2f2",
      },
      "textInfo": {
        ":disabled": {
          "background": "none",
          "color": "#7dd3fc",
        },
        ":focus": {
          "background": "#f0f9ff",
          "boxShadow": "0px 0px 0px 4px #e0f2fe",
        },
        ":hover": {
          "background": "#e0f2fe",
        },
        "background": "#f0f9ff",
      },
      "textInherit": {
        ":disabled": {
          "background": "none",
          "color": "#cbd5e1",
        },
        ":focus": {
          "background": "#f8fafc",
          "boxShadow": "0px 0px 0px 4px #f1f5f9",
        },
        ":hover": {
          "background": "#f1f5f9",
        },
        "background": "#f8fafc",
      },
      "textPrimary": {
        ":disabled": {
          "background": "none",
          "color": "#93c5fd",
        },
        ":focus": {
          "background": "#eff6ff",
          "boxShadow": "0px 0px 0px 4px #dbeafe",
        },
        ":hover": {
          "background": "#dbeafe",
        },
        "background": "#eff6ff",
      },
      "textSuccess": {
        ":disabled": {
          "background": "none",
          "color": "#6ce9a6",
        },
        ":focus": {
          "background": "#ecfdf3",
          "boxShadow": "0px 0px 0px 4px #d1fadf",
        },
        ":hover": {
          "background": "#d1fadf",
        },
        "background": "#ecfdf3",
      },
      "textWarning": {
        ":disabled": {
          "background": "none",
          "color": "#fdba74",
        },
        ":focus": {
          "background": "#fff7ed",
          "boxShadow": "0px 0px 0px 4px #ffedd5",
        },
        ":hover": {
          "background": "#ffedd5",
        },
        "background": "#fff7ed",
      },
    },
  },
  "MuiCard": {
    "styleOverrides": {
      "root": {
        "boxShadow": "0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)",
        "padding": "20px 16px",
      },
    },
  },
  "MuiChip": {
    "styleOverrides": {
      "colorError": {
        "&.MuiChip-filled": {
          "background": "#fef2f2",
          "color": "#b91c1c",
        },
        "&.MuiChip-outlined": {
          "borderColor": "#b91c1c",
          "color": "#b91c1c",
        },
      },
      "colorInfo": {
        "&.MuiChip-filled": {
          "background": "#f0f9ff",
          "color": "#0369a1",
        },
        "&.MuiChip-outlined": {
          "borderColor": "#0369a1",
          "color": "#0369a1",
        },
      },
      "colorSuccess": {
        "&.MuiChip-filled": {
          "background": "#ecfdf3",
          "color": "#027a48",
        },
        "&.MuiChip-outlined": {
          "borderColor": "#027a48",
          "color": "#027a48",
        },
      },
      "colorWarning": {
        "&.MuiChip-filled": {
          "background": "#fff7ed",
          "color": "#c2410c",
        },
        "&.MuiChip-outlined": {
          "borderColor": "#c2410c",
          "color": "#c2410c",
        },
      },
      "filled": {
        "background": "#f1f5f9",
        "color": "#334155",
      },
      "filledPrimary": {
        "background": "#eff6ff",
        "color": "#1d4ed8",
      },
      "outlined": {
        "borderColor": "#334155",
        "borderWidth": "2px",
        "color": "#334155",
      },
      "outlinedPrimary": {
        "borderColor": "#1d4ed8",
        "color": "#1d4ed8",
      },
      "sizeMedium": {
        "fontSize": "14px",
        "fontWeight": 500,
        "padding": "2px 0px",
      },
      "sizeSmall": {
        "fontSize": "12px",
      },
    },
  },
  "MuiCircularProgress": {
    "defaultProps": {
      "color": "inherit",
      "disableShrink": true,
    },
  },
  "MuiFormControl": {
    "defaultProps": {
      "fullWidth": true,
    },
    "styleOverrides": {
      "root": {
        "& .MuiFormHelperText-root": {
          "fontSize": "14px",
          "marginTop": "6px",
          "transform": "translate(-14px)",
        },
        "&.Mui-error": {
          "& fieldset,& fieldset:hover": {
            "border": "1px solid #fca5a5",
          },
        },
        "marginTop": "16px",
      },
    },
  },
  "MuiFormControlLabel": {
    "styleOverrides": {
      "root": {
        "&.MuiFormControlLabel-label": {
          "fontWeight": 400,
        },
      },
    },
  },
  "MuiFormLabel": {
    "styleOverrides": {
      "root": {
        "& .MuiFormLabel-asterisk": {
          "color": "#f87171",
        },
        "color": "inherit",
        "fontSize": "14px",
        "marginBottom": "4px",
      },
    },
  },
  "MuiInput": {
    "defaultProps": {
      "disableUnderline": true,
    },
    "styleOverrides": {
      "root": {
        "& svg.MuiSelect-icon": {
          "& polyline": {
            "strokeWidth": "20px",
          },
          "fontSize": "20px",
          "right": "12px",
        },
        "&.Mui-error": {
          "& svg": {
            "color": "#ef4444",
          },
          "&.Mui-focused": {
            "border": "1px solid #fca5a5",
            "boxShadow": "0px 0px 0px 3px #fee2e2",
          },
          "border": "1px solid #fca5a5",
        },
        "&.Mui-focused": {
          "border": "1px solid #93c5fd",
          "boxShadow": "0px 0px 0px 3px #dbeafe",
        },
        "&.MuiInputBase-multiline": {
          "height": "auto",
        },
        "&::placeholder": {
          "color": "#64748b",
        },
        "&:not([datatype='select-input'])": {},
        "backgroundColor": "#fff",
        "border": "1px solid #cbd5e1",
        "borderRadius": "8px",
        "height": "40px",
        "padding": "8px 12px",
        "transition": ".25s",
        "width": "100%",
      },
    },
  },
  "MuiInputBase": {
    "styleOverrides": {
      "root": {
        "&[datatype='select-input']": {
          "marginTop": "0px",
        },
      },
    },
  },
  "MuiInputLabel": {
    "defaultProps": {
      "shrink": true,
    },
    "styleOverrides": {
      "root": {
        "& .MuiFormLabel-asterisk": {
          "color": "#f87171",
        },
        "color": "inherit",
        "marginBottom": "4px",
      },
      "shrink": {
        "fontSize": "14px",
        "transform": "scale(1) translate(0px, -8px)",
      },
    },
  },
  "MuiModal": {
    "styleOverrides": {
      "root": {
        "& .btn-close": {
          "position": "absolute",
          "right": "12px",
          "top": "12px",
        },
        "& > .MuiBox-root": {
          "background": "#fff",
          "borderRadius": "12px",
          "boxShadow": "0px 24px 48px -12px rgba(16, 24, 40, 0.18)",
          "maxWidth": "900px",
          "minHeight": "100px",
          "minWidth": "320px",
          "outline": "none",
          "position": "relative",
          "width": "100%",
        },
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      },
    },
  },
  "MuiPaginationItem": {
    "styleOverrides": {
      "root": {
        "&.Mui-selected": {
          "background": "#eff6ff",
          "color": "#3b82f6",
          "fontWeight": 500,
          "pointerEvents": "none",
        },
      },
    },
  },
  "MuiPaper": {
    "styleOverrides": {
      "root": {
        "&.select-dropdown": {
          "border": "1px solid #EAECF0",
          "borderRadius": "8px",
          "boxShadow": "0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)",
          "padding": "4px 10px",
        },
        "&.select-dropdown li.MuiMenuItem-root": {
          "&:active": {
            "backgroundColor": "#bfdbfe",
          },
          "borderRadius": "6px",
          "height": "44px",
          "padding": "0px 20px",
        },
        "&.select-dropdown li.MuiMenuItem-root[data-value=' ']": {
          "display": "none",
        },
        "&.select-dropdown ul": {
          "display": "flex",
          "flexDirection": "column",
          "gap": "4px",
        },
      },
    },
  },
  "MuiSelect": {
    "defaultProps": {
      "IconComponent": {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "MenuProps": {
        "PaperProps": {
          "className": "select-dropdown",
        },
      },
      "datatype": "select-input",
      "disableUnderline": true,
      "variant": "standard",
    },
    "styleOverrides": {
      "standard": {
        ":focus": {
          "backgroundColor": "#fff",
        },
      },
    },
  },
  "MuiSnackbar": {
    "defaultProps": {
      "anchorOrigin": {
        "horizontal": "right",
        "vertical": "bottom",
      },
    },
    "styleOverrides": {
      "root": {
        "& .MuiPaper-root": {
          "background": "#fff",
          "color": "#334155",
        },
      },
    },
  },
  "MuiTableBody": {
    "styleOverrides": {
      "root": {
        "& .MuiTableRow-root .MuiTableCell-root": {
          "color": "#334155",
        },
        "& .MuiTableRow-root .MuiTableCell-root svg": {
          "width": "17px",
        },
        "&.hoverable .MuiTableRow-root:hover": {
          "& .MuiTableCell-root": {
            "color": "#0f172a",
          },
          "backgroundColor": "#f0f9ff",
          "cursor": "pointer",
        },
      },
    },
  },
  "MuiTableCell": {
    "styleOverrides": {
      "root": {
        "& .MuiIconButton-root.MuiButtonBase-root": {
          "& svg": {
            "color": "#64748b",
            "left": "50%",
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%) scale(.9)",
          },
          "borderRadius": "4px",
          "height": "28px",
          "padding": "0",
          "width": "28px",
        },
        "& .MuiIconButton-root.MuiButtonBase-root:hover": {
          "& svg": {
            "color": "#0ea5e9",
          },
          "background": "#e0f2fe",
        },
        "& a": {
          "color": "#334155",
        },
        "borderBottom": "1px solid #e2e8f0",
        "fontSize": "14px",
        "fontWeight": 500,
        "height": "54px",
        "padding": "0px 16px",
      },
    },
  },
  "MuiTableHead": {
    "styleOverrides": {
      "root": {
        "& .MuiTableRow-root .MuiTableCell-root": {
          "borderTop": "1px solid #e2e8f0",
          "color": "#334155",
          "fontSize": "12px",
        },
      },
    },
  },
  "MuiTextField": {
    "styleOverrides": {
      "root": {
        "&.date-picker": {
          "& .MuiInputBase-root.MuiOutlinedInput-root": {
            "& button svg": {
              "& polyline": {
                "strokeWidth": "20px",
              },
              "fontSize": "20px",
              "marginRight": "6px",
            },
            "& input": {
              "paddingLeft": 0,
            },
            "&.Mui-focused": {
              "& fieldset,&:hover fieldset": {
                "border": "none",
              },
              "border": "1px solid #93c5fd",
              "boxShadow": "0px 0px 0px 3px #dbeafe",
            },
            "&::placeholder": {
              "color": "#64748b",
            },
            "&:hover fieldset": {
              "border": "1px solid #cbd5e1",
            },
            "backgroundColor": "#fff",
            "borderRadius": "8px",
            "fontSize": "16px",
            "height": "40px",
            "padding": "8px 12px",
            "transition": ".25s",
            "width": "100%",
          },
        },
      },
    },
  },
  "MuiTooltip": {
    "defaultProps": {
      "arrow": true,
      "placement": "top",
    },
    "styleOverrides": {
      "arrow": {
        "color": "#fff",
      },
      "tooltip": {
        "background": "#fff",
        "borderRadius": "8px",
        "boxShadow": "0px 12px 16px -4px rgba(16, 24, 40, 0.08), 0px 4px 6px -2px rgba(16, 24, 40, 0.03)",
        "color": "#64748b",
        "padding": "8px 12px",
      },
    },
  },
}
`;
